client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -126,11 +129,32 @@ tie.factory('PythonCodePreprocessorService', [
         while (true) {
           var codeToAdd = '';
           lastFunctionNameLocation = code.indexOf(
-            functionName, lastFunctionNameLocation);
+            functionNameWithoutParen, lastFunctionNameLocation);
           if (lastFunctionNameLocation === -1) {
             break;
           }
           if (lastFunctionNameLocation !== functionNameStart) {
+            var whitespaceCheckLocation = (
+              lastFunctionNameLocation + functionNameWithoutParen.length);
+            // We assume we found the matched function, but we might have found
+            // a similarly-named other function / variable. User-inputted code
+            // can be ... tricky.
+            var isMatchedFunction = true;
+            // We can safely assume there's a left paren because we matched it
+            // with the regular expression.
+            while (code[whitespaceCheckLocation] != '(') {
+              var isValidWhitespace = VALID_WHITESPACE_REGEX.test(
```

Comment:
I also added a specific _checkMatchedFunctionForWhitespace method to break it down, but I briefly forgot to break out the isWhitespaceChar, so I'll do that now.